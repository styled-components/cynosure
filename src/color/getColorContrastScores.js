// @flow
import getColorContrast from './getColorContrast'

import type { ContrastScores } from '../types/color'

/**
 * Returns the contrast scores between two colors based on the
 * [contrast scores recommended by W3](https://www.w3.org/TR/WCAG20/#visual-audio-contrast).
 *
 * @example
 * const scores = getColorContrastScores('#444', '#fff');
 */
export default function getColorContrastScores(
  color1: string,
  color2: string,
): ContrastScores {
  const contrastRatio = getColorContrast(color1, color2)
  return {
    AA: contrastRatio >= 4.5,
    AALarge: contrastRatio >= 3,
    AAA: contrastRatio >= 7,
    AAALarge: contrastRatio >= 4.5,
  }
}
