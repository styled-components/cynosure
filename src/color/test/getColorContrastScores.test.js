// @flow
import getColorContrastScores from '../getColorContrastScores'

describe('getColorContrastScores', () => {
  it('should return the color contrast scores of two hex colors', () => {
    expect(getColorContrastScores('#444', '#fff')).toMatchSnapshot()
  })

  it('should return the color contrast scores of two 8-digit hex colors', () => {
    expect(getColorContrastScores('#6564CDB3', '#ffffff')).toMatchSnapshot()
  })

  it('should return the color contrast scores of two 4-digit hex colors', () => {
    expect(getColorContrastScores('#0f08', '#000')).toMatchSnapshot()
  })

  it('should return the color contrast scores of two rgba colors', () => {
    expect(
      getColorContrastScores('rgba(101,100,205,0.7)', 'rgba(0,0,0,1)'),
    ).toMatchSnapshot()
  })

  it('should return the color contrast scores of two rgb colors', () => {
    expect(
      getColorContrastScores('rgb(204,205,100)', 'rgb(0,0,0)'),
    ).toMatchSnapshot()
  })

  it('should return the color contrast scores of two hsla colors', () => {
    expect(
      getColorContrastScores(
        'hsla(250, 100%, 50%, 0.2)',
        'hsla(0, 100%, 100%, 1)',
      ),
    ).toMatchSnapshot()
  })

  it('should return the color contrast scores of two hsl colors', () => {
    expect(
      getColorContrastScores('hsl(0, 100%, 50%)', 'hsl(0, 100%, 100%)'),
    ).toMatchSnapshot()
  })

  it('should return the color contrast scores of two named CSS colors', () => {
    expect(getColorContrastScores('papayawhip', 'black')).toMatchSnapshot()
  })
})
