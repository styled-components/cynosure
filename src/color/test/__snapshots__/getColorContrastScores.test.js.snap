// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getColorContrastScores should return the color contrast scores of two 4-digit hex colors 1`] = `
Object {
  "AA": true,
  "AAA": true,
  "AAALarge": true,
  "AALarge": true,
}
`;

exports[`getColorContrastScores should return the color contrast scores of two 8-digit hex colors 1`] = `
Object {
  "AA": true,
  "AAA": false,
  "AAALarge": true,
  "AALarge": true,
}
`;

exports[`getColorContrastScores should return the color contrast scores of two hex colors 1`] = `
Object {
  "AA": true,
  "AAA": true,
  "AAALarge": true,
  "AALarge": true,
}
`;

exports[`getColorContrastScores should return the color contrast scores of two hsl colors 1`] = `
Object {
  "AA": false,
  "AAA": false,
  "AAALarge": false,
  "AALarge": true,
}
`;

exports[`getColorContrastScores should return the color contrast scores of two hsla colors 1`] = `
Object {
  "AA": true,
  "AAA": true,
  "AAALarge": true,
  "AALarge": true,
}
`;

exports[`getColorContrastScores should return the color contrast scores of two named CSS colors 1`] = `
Object {
  "AA": true,
  "AAA": true,
  "AAALarge": true,
  "AALarge": true,
}
`;

exports[`getColorContrastScores should return the color contrast scores of two rgb colors 1`] = `
Object {
  "AA": true,
  "AAA": true,
  "AAALarge": true,
  "AALarge": true,
}
`;

exports[`getColorContrastScores should return the color contrast scores of two rgba colors 1`] = `
Object {
  "AA": false,
  "AAA": false,
  "AAALarge": false,
  "AALarge": true,
}
`;
